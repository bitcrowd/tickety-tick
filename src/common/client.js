import ky from 'ky';

const credentials = 'include';

const headers = {
  'content-type': 'application/json; charset=utf-8',
  accept: 'application/json',
};

function client(base, options = {}) {
  return ky.extend({
    credentials, headers, prefixUrl: base, timeout: 1000, ...options,
  });
}

export default client;
