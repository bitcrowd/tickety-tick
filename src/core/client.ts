import type { Options } from "ky";
import ky from "ky";

const credentials = "include";

const headers = {
  "content-type": "application/json; charset=utf-8",
  accept: "application/json",
};

const timeout = 1000;

function client(base: string, options: Partial<Options> = {}) {
  return ky.extend({
    prefixUrl: base,
    credentials,
    headers,
    timeout,
    ...options,
  });
}

export default client;
