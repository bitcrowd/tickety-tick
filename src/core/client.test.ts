import ky from "ky";

import client from "./client";

jest.mock("ky");

describe("client", () => {
  const mock = { ky: true };

  beforeEach(() => {
    (ky.extend as jest.Mock).mockReturnValue(mock);
  });

  afterEach(() => {
    (ky.extend as jest.Mock).mockReset();
  });

  it("creates a new client with default values", () => {
    const instance = client("https://example.io/api");
    expect((ky.extend as jest.Mock).mock.calls).toMatchSnapshot();
    expect(instance).toBe(mock);
  });

  it("sets custom options", () => {
    const instance = client("https://example.io/api", {
      credentials: "same-origin",
      timeout: 3000,
    });
    expect((ky.extend as jest.Mock).mock.calls).toMatchSnapshot();
    expect(instance).toBe(mock);
  });
});
