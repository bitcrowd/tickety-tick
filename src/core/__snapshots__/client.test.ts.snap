// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`client creates a new client with default values 1`] = `
[
  [
    {
      "credentials": "include",
      "headers": {
        "accept": "application/json",
        "content-type": "application/json; charset=utf-8",
      },
      "prefixUrl": "https://example.io/api",
      "timeout": 1000,
    },
  ],
]
`;

exports[`client sets custom options 1`] = `
[
  [
    {
      "credentials": "same-origin",
      "headers": {
        "accept": "application/json",
        "content-type": "application/json; charset=utf-8",
      },
      "prefixUrl": "https://example.io/api",
      "timeout": 3000,
    },
  ],
]
`;
